build.sh

```diff
@@ -157,7 +157,16 @@ check_uuid_binary(){
 }
 
 run_build() {
-  /home/shippable/cexec/dist/main/main
+  local use_default_arch=true
+  if [ ! -z "$SHIPPABLE_NODE_ARCHITECTURE" ]; then
+    if [ ! -f /home/shippable/cexec/dist/$SHIPPABLE_NODE_ARCHITECTURE/linux/main/main ]; then
+      use_default_arch=false
+      /home/shippable/cexec/dist/$SHIPPABLE_NODE_ARCHITECTURE/linux/main/main
+    fi
```

Comment:
add an else statement here with the statement that arch is present or something. we should be very clear on what's going on in workflows like these so that there is no confusion later when we're debugging this stuff